src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/NuGetRestoreCycleDetector.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore
+{
+    internal sealed class NuGetRestoreCycleDetector
+    {
+        private readonly object _lock;
+        private readonly Queue<byte[]> _values;
```

Comment:
>Queue [](http://example.com/codeflow?start=25&length=5)

Why are we using a queue here and not some data structure with O(1) lookup? I expected this to be a hash table of hashes. A queue doesn't make a lot of sense in my mind. Am I missing some info?

Reply:
I added the hash _lookupTable to improve the search to O(1)

Reply:
I'm still not convinced a queue is the correct data structure here. Can you please call me to discuss as requested?